cmd/syncthing/main.go

```diff
@@ -347,7 +348,13 @@ func main() {
 			// Use leveldb database locks to protect against concurrent upgrades
 			_, err = leveldb.OpenFile(locations[locDatabase], &opt.Options{OpenFilesCacheCapacity: 100})
 			if err != nil {
-				l.Fatalln("Cannot upgrade, database seems to be locked. Is another copy of Syncthing already running?")
+				l.Warnln("The database seems to be locked. Trying to inform the running copy of Syncthing about the upgrade")
```

Comment:
Probably shouldn't print a warning here if it's going to succeed. Just report the final error, whatever it was ("talked to syncthing, it said no" or "tried to upgrade, but ...").